source/common/network/utility.cc

```diff
@@ -240,16 +240,32 @@ Address::InstanceConstSharedPtr Utility::getLocalAddress(const Address::IpVersio
 
 bool Utility::isLocalConnection(const Network::ConnectionSocket& socket) {
   const auto& remote_address = socket.remoteAddress();
-  // Before calling getifaddrs, verify the obvious checks.
-  // Note that there are corner cases, where remote and local address will be the same
-  // while the client is not actually local. Example could be an iptables intercepted
-  // connection. However, this is a rare exception and such assumption results in big
-  // performance optimization.
+  // Before calling getifaddrs, verify the obvious checks. These are local:
```

Comment:
Updated comment per @mattklein123 's request.
I removed note about corner case as I feel that if something like iptables has made Envoy see the same remote and local IP addresses, then we are expected to treat it as local; it's not a corner case. I'm also fine with putting it back.